pkg/daemon/ceph/client/filesystem.go

```diff
@@ -378,6 +379,21 @@ func WaitForNoStandbys(context *clusterd.Context, clusterInfo *ClusterInfo, retr
 	return nil
 }
 
+func filesystemHasStandby(dump *MDSDump, fsName string) bool {
+	for _, standby := range dump.Standbys {
+		// The mds dump does not explicitly return the name of the filesystem that the
+		// daemon belongs to, so the matching to the filesystem name is based on the mds daemon name
+		// with a regular expression comparison with the expected suffix.
+		// For example, if the filesystem is "myfs", the standby name may be "myfs-a" or "myfs-b".
+		matchString := fmt.Sprintf("^%s-[a-z]{1}$", fsName)
```

Comment:
whats if one filesystem is

filesystemName1 := myfs
filesystemName2 := myfs-a

can it do a wrong match if it will compare `myfs`?
can we try a unit test for this

Reply:
The regex will handle those cases of only being an exact match at the beginning (with `^`) and and the end (with `$`), so that substrings will not be matched. This is also covered by the unit test [here](https://github.com/rook/rook/pull/14952/files#diff-306c3a98346ecc31a6003aaaac25d81777818d1c86c95243a3f67518fc51306aR530), which only has a single unexpected char at the end, and does not match.